### ISSUE TITLE
About training on ImageNet

### ISSUE THREAD

**Issue by LMMMEng** [MzeroMiko/VMamba#165]
Hi，

I noticed in your training log (imagenet) that the training loss is normal, but the loss in the test phase is nan in many epochs, I also encountered this problem, could you please tell me how to solve it?

**Comment by MzeroMiko** [comment]
The reason why the test loss is nan is that when inferencing with `torch.no_grad()`, we set all the data in float16 in that period of training. We fixed that later.

**Comment by LMMMEng** [comment]
Does that mean this is caused by using AMP when inferencing?

**Comment by MzeroMiko** [comment]
In our model, even if you are using amp, we still forces some part of data to be in float32

**Comment by LMMMEng** [comment]
I see. I still don't understand why training loss is normal, and test loss becomes nan, according to your code,  it seems they all use amp.

**Comment by Journey7331** [comment]
Hi, @MzeroMiko @LMMMEng I notice this `grad_norm 1.5292 (inf)` in the log `vssm_tiny_0230.txt` that MzeroMiko provided, and actually i also encounter this `inf` and `loss=nan` in my training, so is there any relation between `inf` and `nan`?

<img width="1679" alt="image" src="https://github.com/MzeroMiko/VMamba/assets/66991472/96725bcb-d328-4114-bc24-465e80595d56">

**Comment by LMMMEng** [comment]
Hi, @Journey7331. It is usually normal for grad to occasionally appear inf, because amp automatically detects this kind of inf and adjusts the loss scale. However, if the gradient continues to appear inf, even if the loss scale is automatically adjusted to a very small value, the gradient may not return to normal, then the loss will be nan. Please feel free to correct me if I'm wrong.
